PROYECTO INTEGRADOR - MINERIA DE DATOS : INSPECCIÓN INICIAL



Importación de librerías y Carga del JSON

In [2]:
import pandas as pd
import os

# Definimos la ruta relativa hacia el JSON original sucio
ruta_json = os.path.join("..", "data", "raw", "streaming_users_dirty.json")

# Cargamos el archivo JSON exigido por la cátedra
df = pd.read_json(ruta_json)

# Inspección inicial de las primeras filas
df.head()

,user_id,age,subscription_plan,monthly_watch_time_mins,country,favorite_genre,last_login_date,customer_support_tickets
0,10000,39,Estándar,805.8,Brasil,Crime,2025-03-04,99
1,10001,37,Estándar,1173.4,Colombia,Crime,2019-04-02,2
2,10002,28,Básico,401.0,Colombia,Crime,2018-04-13,0
3,10003,43,Básico,62.4,Uruguay,Thriller,2021-01-31,0
4,10004,51,Básico,477.8,Perú,Thriller,2020-09-30,1


Dimensiones del Dataset
Acá le mostramos al profesor cuántos usuarios y cuántas variables (columnas) tiene el archivo original.

In [3]:
# Ver el total de filas y columnas (Formato: (filas, columnas))
print(f"Cantidad de filas: {df.shape[0]}")
print(f"Cantidad de columnas: {df.shape[1]}")

Cantidad de filas: 8160
Cantidad de columnas: 8


Estructura y Tipos de Datos (info)

In [4]:
df.tail()

,user_id,age,subscription_plan,monthly_watch_time_mins,country,favorite_genre,last_login_date,customer_support_tickets
8155,10923,23,Premium,1161.4,Colombia,Romance,2023-05-15,0
8156,16525,27,Básico,436.2,Uruguay,Documental,2018-09-06,4
8157,11222,13,Estándar,1321.8,México,Documental,2019-02-08,0
8158,15613,38,Estándar,835.7,Brasil,Drama,2022-02-05,0
8159,16912,25,Estándar,1468.7,Argentina,Romance,2022-08-12,3


In [16]:
# Inspección técnica de los tipos de datos por columna
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 8160 entries, 0 to 8159
Data columns (total 8 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   user_id                   8160 non-null   int64  
 1   age                       8160 non-null   int64  
 2   subscription_plan         8160 non-null   str    
 3   monthly_watch_time_mins   7967 non-null   float64
 4   country                   8160 non-null   str    
 5   favorite_genre            7920 non-null   str    
 6   last_login_date           7840 non-null   str    
 7   customer_support_tickets  8160 non-null   int64  
dtypes: float64(1), int64(3), str(4)
memory usage: 756.7 KB


Identificación de Datos Faltantes (Nulos)
acá medimos el tamaño del problema detectando cuántos nulos hay por columna.

In [5]:
df.dtypes

user_id                       int64
age                           int64
subscription_plan               str
monthly_watch_time_mins     float64
country                         str
favorite_genre                  str
last_login_date                 str
customer_support_tickets      int64
dtype: object

In [6]:
# Contamos cuántos valores nulos (missing values) tiene cada variable
valores_nulos = df.isnull().sum()
print("Cantidad de valores nulos por columna:")
print(valores_nulos[valores_nulos > 0]) # Muestra solo las que tienen nulos

Cantidad de valores nulos por columna:
monthly_watch_time_mins    193
favorite_genre             240
last_login_date            320
dtype: int64


In [7]:
df.describe(include='all')

,user_id,age,subscription_plan,monthly_watch_time_mins,country,favorite_genre,last_login_date,customer_support_tickets
count,8160.000000,8160.000000,8160,7967.000000,8160,7920,7840,8160.000000
unique,NaN,NaN,15,NaN,26,28,3062,NaN
top,NaN,NaN,Básico,NaN,Brasil,Comedia,2026-15-03,NaN
freq,NaN,NaN,3450,NaN,1132,1112,20,NaN
mean,13995.433824,34.096814,NaN,1107.346153,NaN,NaN,NaN,1.800980
std,2310.810660,14.511304,NaN,5310.442604,NaN,NaN,NaN,11.334969
min,10000.000000,-5.000000,NaN,-120.000000,NaN,NaN,NaN,-1.000000
25%,11987.750000,25.000000,NaN,489.200000,NaN,NaN,NaN,0.000000
50%,13998.500000,33.000000,NaN,757.400000,NaN,NaN,NaN,1.000000
75%,15997.250000,42.000000,NaN,1045.700000,NaN,NaN,NaN,1.000000


Explicación de qué hace: Este comando calcula automáticamente las estadísticas de resumen para todas las columnas que contienen números (age, monthly_watch_time_mins y customer_support_tickets). Te mostrará cuántos datos hay, el promedio, el mínimo, el máximo y los percentiles.

In [8]:
df.value_counts()

user_id  age  subscription_plan  monthly_watch_time_mins  country    favorite_genre  last_login_date  customer_support_tickets
10037    33   Básico             611.0                    Colombia   Documental      2019-09-29       2                           2
10052    25   Básico             465.7                    Colombia   Acción          2022-03-31       1                           2
10117    29   Estándar           713.9                    Brasil     Documental      2020-12-20       1                           2
10128    19   Básico             638.7                    Argentina  Drama           2020-06-17       1                           2
10156    43   Básico             592.8                    Brasil     Romance         2021-10-25       0                           2
                                                                                                                                 ..
11796    21   Premium            1427.8                   Perú       Drama       

In [9]:
df['country'].value_counts()

country
Brasil        1132
Chile         1132
México        1129
Uruguay       1124
Perú          1120
Colombia      1116
Argentina     1087
colombia        27
méxico          25
uruguay         24
Brazil          21
COL             19
CHL             18
URY             17
MEX             16
Chile           16
argentina       16
PER             16
chile           15
Mexico          15
Peru            15
BRA             15
brasil          13
perú            12
ARG             10
Argentina       10
Name: count, dtype: int64

In [9]:
df.describe()

,user_id,age,monthly_watch_time_mins,customer_support_tickets
count,8160.000000,8160.000000,7967.000000,8160.000000
mean,13995.433824,34.096814,1107.346153,1.800980
std,2310.810660,14.511304,5310.442604,11.334969
min,10000.000000,-5.000000,-120.000000,-1.000000
25%,11987.750000,25.000000,489.200000,0.000000
50%,13998.500000,33.000000,757.400000,1.000000
75%,15997.250000,42.000000,1045.700000,1.000000
max,17999.000000,150.000000,99999.000000,150.000000


In [10]:
df.isnull().sum()


user_id                       0
age                           0
subscription_plan             0
monthly_watch_time_mins     193
country                       0
favorite_genre              240
last_login_date             320
customer_support_tickets      0
dtype: int64

In [11]:
(df.isnull().sum()/len(df))*100

user_id                     0.000000
age                         0.000000
subscription_plan           0.000000
monthly_watch_time_mins     2.365196
country                     0.000000
favorite_genre              2.941176
last_login_date             3.921569
customer_support_tickets    0.000000
dtype: float64

In [13]:
df.duplicated()

0       False
1       False
2       False
3       False
4       False
        ...  
8155     True
8156     True
8157    False
8158    False
8159     True
Length: 8160, dtype: bool

In [14]:
df.duplicated().sum()

np.int64(126)

In [15]:
df.nunique()

user_id                     8000
age                           69
subscription_plan             15
monthly_watch_time_mins     5788
country                       26
favorite_genre                28
last_login_date             3062
customer_support_tickets       9
dtype: int64

In [16]:
df["age"].value_counts()

age
13    391
31    273
36    272
32    270
29    269
     ... 
69      2
71      2
76      1
74      1
70      1
Name: count, Length: 69, dtype: int64

In [17]:
df.sample()

,user_id,age,subscription_plan,monthly_watch_time_mins,country,favorite_genre,last_login_date,customer_support_tickets
7933,17933,43,Estándar,1094.1,México,Romance,2025-02-02,1


### Evidencia Observada - Variables Numéricas:
* **Edad (`age`):** Se observa que el rango de edad se encuentra entre [mínimo] y [máximo] años.
* **Tiempo de Visualización (`monthly_watch_time_mins`):** El promedio de visualización mensual es de [promedio] minutos. Se evidencia la presencia de valores faltantes (nulos) dado que el conteo es menor al total de filas.
* **Tickets de Soporte (`customer_support_tickets`):** El número de reclamos por usuario va desde [mínimo] hasta [máximo].

## Evidencia Detectada para la Etapa de Calidad y Limpieza
1. **Datos Faltantes:** Se identificaron valores nulos en `favorite_genre` (240 nulos), `monthly_watch_time_mins` (193 nulos) y `last_login_date` (320 nulos). Requerirán una estrategia de imputación o eliminación justificada.
2. **Tipos de Datos Erróneos:** La variable `last_login_date` se encuentra registrada como `object` (texto). Deberá ser convertida a tipo `datetime` para permitir análisis temporales.

Preguntas de Análisis Propuestas
* ¿Existe alguna relación entre la edad de los usuarios (`age`) y el género de contenido preferido (`favorite_genre`)?
* ¿El tiempo de visualización mensual (`monthly_watch_time_mins`) varía significativamente según el tipo de plan contratado (`subscription_plan`)?
* ¿Los usuarios que abren una mayor cantidad de tickets de soporte técnico (`customer_support_tickets`) presentan un menor tiempo de uso en la plataforma?
--